subs/pantry/src/Pantry/Archive.hs

```diff
@@ -443,3 +468,15 @@ takeSubdir subdir = mapMaybe $ \(fp, a) -> do
   where
     splitDirs = List.dropWhile (== ".") . filter (/= "") . T.splitOn "/"
     subdirs = splitDirs subdir
+
+getCabalBS :: forall env. (HasPantryConfig env, HasLogFunc env, HasProcessContext env)
+           => ByteString -- Hpack's content
+           -> RIO env ByteString
+getCabalBS hpackBs = withSystemTempDirectory "hpack-repo" $ \tmpdir -> withWorkingDir tmpdir $ do
```

Comment:
couldn't we use a more explicit name like `hpackToCabal`? Also we dealing with archives here, what's the point of "repo" in directory name?

Reply:
Renamed it to `hpackToCabal`.